internal/server/websocketServer.go

```diff
@@ -217,7 +284,7 @@ func (s *WebsocketServer) processResponse(conn *websocket.Conn, diceDBCmd *cmd.D
 		err = resp.EvalResponse.Error
 	default:
 		slog.Debug("Unsupported response type")
-		if err := WriteResponseWithRetries(conn, []byte("error: 500 Internal Server Error"), maxRetries); err != nil {
+		if err := s.writeResponseWithRetries(conn, []byte("error: 500 Internal Server Error"), maxRetries); err != nil {
```

Comment:
What happens when multiple queries response need to sent back to different client at the same time? How the current lock will handle those scenario?